pkg/schema/definitions/handler.go

```diff
@@ -46,24 +38,40 @@ type schemaDefinitionHandler struct {
 	schemaToModel map[string]string
 }
 
+// Refresh writeLocks and updates the cache with new schemaDefinitions. Will result in a call to kubernetes to retrieve
+// the openAPI schemas.
+func (s *SchemaDefinitionHandler) Refresh() error {
+	s.Lock()
+	defer s.Unlock()
```

Comment:
IIUC this locks readers for the whole duration of the OpenAPI schema refresh.

Could we lock later, ideally only when all OpenAPI calls and network I/O are finished, so that most of the refresh won't block?

Reply:
I think that you are correct. In the previous implementation I think this had value since the refresh was initiated by the refresh, so the change you are talking about could have resulted in multiple calls to the openapi endpoints on the k8s side. 

However, since the refresh is no longer user-initiated I think the approach that you are talking about will work much better. I'll swap this around.

Reply:
Change made. This also allowed me to clean up some areas around the function calls/interfaces - please let me know if you see anything else here.